pkg/cmd/push/cmd.go

```diff
@@ -58,17 +62,30 @@ func (opts *pushOptions) complete(ctx context.Context, args []string) error {
 	}
 	opts.configHome = configHome
 
-	modelRef, extraTags, err := util.ParseReference(args[0])
+	srcRef, extraTags, err := util.ParseReference(args[0])
 	if err != nil {
-		return fmt.Errorf("failed to parse reference: %w", err)
-	}
-	if modelRef.Registry == "localhost" {
-		return fmt.Errorf("registry is required when pushing")
+		return fmt.Errorf("failed to parse reference %s: %w", args[0], err)
 	}
 	if len(extraTags) > 0 {
 		return fmt.Errorf("reference cannot include multiple tags")
```

Comment:
now that there are 2 references, we should qualify this one as source reference similiar to the dest one.

Reply:
I didn't want to complicate the wording too much (so we do distinguish them by labelling the target `target`). If specified with one argument, calling it the source reference doesn't make a lot of sense to me, as it's both the source and destination.